File: dataset/human/c/file_3540.c
#include<stdio.h>
int P[100];
int A[101];
int main(void){
    int n;
    int i,j,ans;

    scanf("%d",&n);
    for(i=0;i<n;i++){
	scanf("%d",&P[i]);
	for(j=P[i];j>=0;j--){
	    A[j]++;
	}
    }

    ans=0;
    for(i=0;i<=100;i++){
	if(A[i]>=i) ans=i;
    }
    printf("%d\n",ans);

    return 0;
}

File: dataset/human/python/file_3540.py
def factorization(n):
    arr = []
    temp = n
    for i in range(2, int(-(-n**0.5//1))+1):
        if temp%i==0:
            cnt=0
            while temp%i==0:
                cnt+=1
                temp //= i
            arr.append([i, cnt])

    if temp!=1:
        arr.append([temp, 1])

    if arr==[]:
        arr.append([n, 1])

    return arr

f=factorization(int(input())) 
ans=0
for list in f:
  if list[0]==1:
    continue
  m=list[1]
  n=int((2*m)**(1/2))
  while True:
    if n*(1+n)>2*m:
      n-=1
      break
    n+=1

  ans+=n
print(ans)
